{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "3ae0d7d0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "from IPython.display import Markdown\n",
    "\n",
    "number = 9\n",
    "\n",
    "#species Yarrowia or Rhodosporidium\n",
    "species = f\"/Yarrowia papers\"\n",
    "pagenumber = 35\n",
    "\n",
    "file_path = \"./Page \"+str(pagenumber)+\"/\"\n",
    "number=str(number)\n",
    "with open(file_path+number+\" Title.txt\", 'r') as file:\n",
    "    Title_text = file.read()\n",
    "with open(file_path+number+\" Abstract.txt\", 'r') as file:\n",
    "    Abstract_text = file.read()\n",
    "with open(file_path+number+\" Method.txt\", 'r', encoding=\"UTF-8\") as file:\n",
    "    Method_text = file.read()\n",
    "with open(file_path+number+\" Result.txt\", 'r', encoding=\"UTF-8\") as file:\n",
    "    Result_text = file.read()\n",
    "with open(file_path+number+\" Table.txt\", 'r', encoding=\"UTF-8\") as file:\n",
    "    Table_text = file.read()    \n",
    "\n",
    "#for papers with no gene edits, use prompt 1 and 3, otherwise, use 1,2,3\n",
    "prompt1 = \"A multi-column spreadsheet from the given text  | experiment number | Strain name | product | product concentration | substrate | substrate concentration | media | time | Mode of Operation|              \"+ Abstract_text+\"    \"+Result_text\n",
    "prompt2 = \"A multi-column spreadsheet from the given text  | Strain name | parent strain | Knocked out genes | expressed genes | promoter | genome integration Y/N | codon optimized |         When output the daughter strain, you need to include the expressed and knockout genes in the parent strain.             \"+ Method_text+\"    \"+Result_text\n",
    "prompt3 = \"A multi-column spreadsheet from the given text  | experiment type | culture volume | culture vessel | substrate | substrate concentration | oxygen condition | nitrogen condition | time | pH | temperature |      \"+ Method_text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "c9a7fa46",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "Remodelling metabolism for high-level resveratrol production in Yarrowia lipolytica"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Title\n",
    "display(Markdown(Title_text))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "476d92bb",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "A multi-column spreadsheet from the given text  | experiment number | Strain name | product | product concentration | substrate | substrate concentration | media | time | Mode of Operation|              Abstract\n",
       "Resveratrol is a polyphenol with numerous applications in food, pharma, and cosmetics. Lack of precursors and low titer are the main problems hindering industrial scale resveratrol production. Based on previous prescreening, expressing the combination of FjTAL, Pc4CL1 and VvSTS achieved the best resveratrol titer. This was further improved to 235.1 mg/L through engineering the shikimic acid pathway, applying a modular enzyme assembly of Pc4CL1 and VvSTS, enhancing p-coumaric acid supply and diverting glycolytic flux toward erythrose-4-phosphate. The titer was increased to 819.1 mg/L following two rounds of multicopy integration of resveratrol biosynthesis and malonyl-CoA supply, respectively. The titer reached 22.5 g/L with a yield on glucose of 65.5 mg/g using an optimum fed-batch strategy in a 5 L bioreactor with morphology control. This research is the highest report on the de novo production of resveratrol in Yarrowia lipolytica and the findings lay a solid foundation for other producing polyphenols.    3.1. Construction of a basal resveratrol-producing strain\n",
       "For the efficient synthesis of resveratrol, a de novo resveratrol biosynthesis pathway was introduced in Y. lipolytica PO1f (Fig. 1A). Six resveratrol synthesis pathways were constructed by combining six genes (FjTAL from Flavobacterium johnsoniaeu; RgTAL from Rhodotorula glutinis; TcTAL from Trichosporon cutaneum; Pc4CL1 from Petroselinum crispum; At4CL1 from Arabidopsis thaliana; VvSTS from Vitis vinifera) from different sources with the strong TEF promoter. Accumulation of p-CA and resveratrol was detected in all engineered strains (Fig. 1B). The resveratrol titer ranged from 0.3 mg/L to 1.4 mg/L after 96 h of fermentation in 24 deep-well plates. The resveratrol titer of FjTAL strains ST413 and ST414 was slightly higher than that of strains containing RgTAL and TcTAL. To investigate whether the precursor l-tyrosine limits the synthesis of resveratrol, 200 mg/L of l-Tyr precursor was added for fermentation, and this increased resveratrol accumulation in all engineered strains, among which strain ST413 (FjTAL-Pc4CL1-VvSTS) reached the highest (4.6 mg/L) (Fig. 1C). Therefore, ST413 was used as the initial strain for subsequent strain construction. These results showed that the combination of FjTAL and Pc4CL1 achieved the highest resveratrol titer. Fig. 1. Biosynthetic pathways of resveratrol in engineered Y. lipolytica. (A) G-6-P, glucose-6-phosphate; F-6-P, fructose-6-phosphate; F-1,6BP, fructose-1,6-bisphosphate； PEP, phosphoenolpyruvate; E4P, erythrose-4-phosphate; DAHP, 3-deoxy-arabino-heptulonate-7-phosphate; EPSP, 5-enolpyruvylshikimate 3-phosphate; CHA, chorismic acid; l-Phe, l-phenylalanine; l-Tyr, l-tyrosine; ARO3, 3-deoxy-7-phosphoheptulonate synthase; ARO4, 3-deoxy-7-phosphoheptulonate synthase; ARO1, pentafunctional AROM polypeptide; ARO7, Chorismite mutase; FPK, phosphoketolase; PTA, phosphotransferase; TAL, tyrosineammonia lyase; 4CL, 4-coumarate-CoA ligase; STS, resveratrol synthase; PAL, phenylalanine ammonia lyase; C4H, trans-cinnamate 4-monooxygenase; AtATR2, threonine-protein kinase ATR2; YlCYB5, cytochrome b5; DGA1, diacylglycerol O-acyltransferase 1; TAG, triacylglycerol; fbr, feedback-resistant. (B) The titer of p-CA and resveratrol without l-tyrosine. (C) The titer of p-CA and resveratrol with 200 mg/L l-tyrosine (l-Tyr).\n",
       "\n",
       "3.2. Engineering the shikimic acid (SA) pathway for improved resveratrol biosynthesis\n",
       "Addition of l-Tyr showed that enhancing the supply of this precursor could increase the efficiency of resveratrol biosynthesis. Therefore, removing feedback inhibition may be an effective method to increase l-Tyr. According to previous studies, the mutant alleles ARO4 and ARO7 are common engineering targets for enhancing l-Tyr to resveratrol production in Y. lipolytica (Saez-Saez et al., 2020) and S. cerevisiae (Li et al., 2016). Allelic mutant genes Aro4K221L and Aro7G139S were constructed under the strong TEF promoter, and the resveratrol titer was significantly increased when either allele was expressed (Fig. 2). Furthermore, when both Aro4K221L and Aro7G139S were expressed in strain ST420 (FjTAL-PC4CL1-VvSTS, Aro4K221L-Aro7G139S), the titer reached 96.2 mg/L after 96 h fermentation in 24-deep well plates. An attempt to further enhance l-Tyr accumulation was made by overexpressing intron-removed YlARO1. Based on strain ST420, accumulation of resveratrol and p-CA reached 111.2 mg/L and 131.2 mg/L, respectively, by overexpressing intron-removed YlARO1.\n",
       "\n",
       "\n",
       "3.3. Expression of a Pc4CL1-VvSTS fusion enhances resveratrol biosynthesis\n",
       "Although SA feedback-resistant inhibition showed a significant increase in resveratrol titer, the precursor also resulted in a high accumulation of p-CA. To further improve the titer of resveratrol, fusion proteins of Pc4CL1 and VvSTS were used to reduce the potential loss of unstable intermediates. In order to achieve a strong priming effect, the strong 8xUASTEFin promoter (see Supplementary Materials) joined with eight upstream activating sequence (UAS) motifs was constructed to drive expression of the fusions. Specifically, VvSTS was fused to the C-terminus of Pc4CL1 under the control of the strong 8xUASTEFin promoter with a rigid (TPTP, TPTPx2, EAAAK, EAAAKx2) or flexible (GGGGS, GGGGSx2, GSG, GSGx2) linking peptide. Fusion proteins were expressed in SA feedback-resistant strains. The results showed that partial linkers could increase resveratrol production. The ST427 strain expressing the E3AK fusion was the highest titer of 149.2 mg/L, but accumulation of p-CA was only 6.1 mg/L in this strain (Fig. 3).\n",
       "\n",
       "\n",
       "Fusing Pc4CL1 and VvSTS increased the titer of resveratrol, but only a small amount of p-CA accumulated, suggesting that the supply of precursor was insufficient. To improve precursor supply, the PAL branch pathway was used to enhance the accumulation of p-CA. AtPAL and AtC4H from A. thaliana, and SmPAL and SmC4H from Silybum marianum were introduced and co-expressed with endogenous YlCYB5 and AtATR2 from A. thaliana to construct the PAL branch pathway in strain ST427. The titer of resveratrol was increased to 175.1 mg/L in the ST450 (AtPAL-AtC4H) strain (Fig. 4A). The enzyme DAHP synthase (ARO3) is feedback-inhibited by l-Phe (Cao et al., 2020). By overexpressing the allelic mutant YlARO3K225L, the titer of resveratrol reached 220.1 mg/L in strain ST452, and 50.2 mg/L of p-CA was accumulated (Fig. 4A).\n",
       "\n",
       "Low accumulation of p-CA may be the main factor limiting the resveratrol titer. It was speculated that factors upstream may restrict carbon flux to the SA pathway. In order to increase carbon flow into the SA pathway, phosphoketolase from Bifidobacterium brebifidum (BbFPK), Bifidobacterium adolescentis (BaFPK) and Clostridium acetobutylicum (CaFPK), and phosphotransferase from Bacillus subtilis (BsPTA) and E. coli (EcPTA) were used to construct the phosphoketolase (PHK) pathway. The PHK pathway was introduced into strain ST452 to increase the supply of precursor E4P (Fig. 4B). To much surprise, the titer of resveratrol was not significantly increased, but p-CA accumulation was relatively high. Strain ST484 in which CaFPK and BsPTA were integrated produced 235.1 mg/L resveratrol, as well as 518.3 mg/L p-CA (Fig. 4C), suggesting the resveratrol biosynthesis and malonyl-CoA supply genes should be further optimised.\n",
       "\n",
       "3.5. Multi-copy integration of key genes improves resveratrol biosynthesis\n",
       "Redistributing carbon metabolism leads to the accumulation of p-CA. Herein, a multi-copy integration strategy was used to improve the downstream metabolic pathway. Two multi-copy site sequences have been identified in Y. lipolytica, 26S rDNA (Lv et al., 2019) and ZETA (Bordes et al., 2007), which have a copy number of 100–200 in the genome. Because Y. lipolytica Po1f is URA3- and LEU-deficient, the auxotroph can be recovered following integration, which is conducive to large-scale industrial fermentation. To achieve higher filtering efficiency, weak PLEU promoters were used to construct URA3 and LEU screening tags. Expression of the Pc4CL1-VvSTS fusion was achieved using 26S rDNA carrying URA3 (Fig. 5A). As shown in Fig. 5B, 72 clones were randomly selected for 24 deep-well plate fermentation, and all strains produced resveratrol and p-CA. Following the first round of 26S rDNA random integration, the highest titer was 486.2 mg/L with strain ST888 (Fig. 5B). Since the uracil deficiency was recovered in the previous 26S rDNA integration process, the LEU marker could be recovered by constructing screening tags in the process of ZETA multi-copy integration. Following the second round of ZETA integration in strain ST888, 72 clones were randomly selected for fermentation, and the highest titer of resveratrol was 632.1 mg/L with strain ST889 (Fig. 5C). In order to further increase the titer of resveratrol, it was necessary to increase the supply of malonyl-CoA. By knocking out DGA1, which is responsible for triacylglycerol (TAG) production, in strain ST889, the titer of resveratrol reached 819.1 mg/L with strain ST890 (Fig. 5C).\n",
       "\n",
       "Malonyl-CoA is also positioned at the main rate-limiting step of resveratrol biosynthesis. In a previous study, YlACC1 overexpression was used to increase the supply of malonyl-CoA, but it did not increase the titer of resveratrol (Saez-Saez et al., 2020). In Y. lipolytica, malonyl-CoA is mainly used for the synthesis of TAG, therefore reducing TAG production may be an effective means of accumulating malonyl-CoA (Tai & Stephanopoulos, 2013). Indeed, resveratrol was increased 1.3-fold (819.1 mg/L) compared with ST889 by knocking out DGA1. These results also indicate that simply overexpression related genes of malonyl-CoA may result in the accumulation of TAG, while weakening TAG accumulation is an effective strategy to enhance malonyl-CoA to improve resveratrol production.\n",
       "\n",
       "3.6. Efficient production of resveratrol in a 5 L bioreactor by morphology control\n",
       "Y. lipolytica is a typical dimorphic yeast (Vandermies et al., 2018). In a previous study, filamentous forms are mainly used to resist to a harsh environment (Liu et al., 2021). Therefore, more efficient resveratrol production with strain ST890 was conducted by carefully controlling cell morphology in a 5 L bioreactor. As shown in Fig. 6A, under optimal conditions (pH 4.5), Y. lipolytica existed in both yeast and filamentous forms, and the resveratrol titer reached 11.3 g/L. At pH 4.5, the effect of DO on yeast morphology was explored by controlling the oxygen concentration, and at 20 % DO the filamentous morphology was scarce, most cells were in the yeast form, and the resveratrol titer reached 15.6 g/L (Fig. 6B). At pH 4.5 and 20 % DO, cell morphology was altered by manipulating the feed flow to control the C/N ratio. When the C/N ratio was 10:1, almost all Y. lipolytica cells were in the yeast form, and the resveratrol titer reached the highest level of 22.5 g/L (Fig. 6C) with no more than 20 mg/L of p-CA accumulated. Fig. 6. Fed-batch fermentation optimisation in a 5 L bioreactor. (A) Strain morphology and fermentation results under different pH conditions (DO 10 %). (B) Strain morphology and fermentation results under different DO conditions (pH 4.5). (C) Strain morphology and fermentation results under different C/N conditions (pH 4.5, DO 20 %). (D) Results of fermentation under optimal conditions (pH 4.5, DO 20 %, C/N 10:1).\n",
       "\n",
       "By optimizing pH, DO and C/N in a 5 L fermenter, Y. lipolytica was maintained in the yeast form, a total of 961.8 g glucose was added during the fermentation of 144 h and fed-batch fermentation of ST890 produced 22.5 g/L (0.28 g/g DCW) resveratrol (Fig. 6D), with a yield on glucose of 65.5 mg/g and a productivity of 0.16 g/L/h. Importantly, this high-level synthesis of resveratrol was achieved using only inexpensive mineral fermentation media, which to some extent reduced lipid accumulation and thus increased the accumulation of malonyl-CoA.\n",
       "\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#prompt 1\n",
    "display(Markdown(prompt1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "b5840063",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "A multi-column spreadsheet from the given text  | Strain name | parent strain | Knocked out genes | expressed genes | promoter | genome integration Y/N | codon optimized |         When output the daughter strain, you need to include the expressed and knockout genes in the parent strain.             2. Materials and methods\n",
       "2.1. Strains and medium\n",
       "The Y. lipolytica Po1f (Lv et al., 2019) was used as the starting host strain. Plasmids were constructed and propagated using E. coli JM109. Cultures of E. coli strains at 37 °C were carried out on lysogeny broth (LB) or agar plates containing ampicillin (100 mg/L). Y. lipolytica was cultured on yeast nitrogen base (YNB) medium (6.74 g/L YNB with or without 0.47 g/L amino acids, 20 g/L glucose) or yeast extract peptone dextrose (YPD) medium (20 g/L glucose, 20 g/L peptone and 10 g/L yeast extract) at 30 °C. For plasmids with LEU tags, Y. lipolytica strains were grown on YNB with uracil dropout. For plasmids with URA tags, Y. lipolytica were grown on YNB with leucine dropout. For the non-deficient defective strain, Y. lipolytica were grown on YNB media.\n",
       "\n",
       "2.2. Construction of vectors and strains\n",
       "The strains and plasmids used in this study are shown in Supplementary Materials. A set of vectors from YaliBricks was used to construct the plasmids needed for genome engineering (Lv et al., 2019). All plasmids were constructed using a Gibson assembly kit (Sangon Biotech, Shanghai, China). Especially, the fusion of Pc4CL1 and VvSTS proteins was constructed by adding homologous arms of linker sequences at both ends. All heterologous genes were codon optimized and synthesized by Sangon Biotech (Sangon Biotech, Shanghai, China). All endogenous genes were amplified from the Y. lipolytica PO1f genome by using a High-fidelity Phusion DNA polymerase (Vazyme, Nanjing, China). All plasmids were confirmed using Sanger sequencing by Sangon Biotech (Sangon Biotech, Shanghai, China). CRISPR-Cas9 system (Wong et al., 2017) was used for single gene integration or knock-out. CRISPR RGEN Tools (https://chopchop.cbu.uib.no/) website was used for the specific guide RNA (sgRNA) design. The transforming plasmids or genome integration of Y. lipolytica PO1f were performed according to the requirements of A Frozen-EZ Yeast Transformation II Kit (Zymol Research, CA, USA). After transformation, the transformed cells were evenly coated on the corresponding selective solid medium and cultivated at 30 °C for 3–5 days.\n",
       "\n",
       "For multi-copy integration, 26S ribosomal DNA (rDNA) and ZETA sequences were copied from the Y. lipolytica genome. The fragments are connected to the upstream and downstream homologous arms of 26S rDNA or ZETA sequence, generating prDNA::(URA3-Pc4CL1-E3AK-VvSTS) and pZETA::(LEU-Pc4CL1-E3AK-VvSTS), respectively. Each vector was linearised to Y. lipolytica transformation.\n",
       "\n",
       "2.3. Strain cultivation and fed-batch fermentation\n",
       "All small-scale fermentation was carried out in 24-well plates. For 24-well plate fermentation, strains were spread on YPD plates and cultivated at 30 °C for 2–3 days. After cultivated, single colony was randomly selected in 24 deep-well plates containing 2 mL of YPD medium and incubated at 30 °C and 220 rpm for 18 h as seed. They were then transferred to new 24-well plates containing 4 mL YPD medium at an initial inoculum of 5 % for fermentation. After 96 h, OD600 was made and samples were collected for high-performance liquid chromatography (HPLC) analysis.\n",
       "\n",
       "For fed-batch fermentation, strains from seed culture preparation were spread on YPD plates and cultivated at 30 °C for 2–3 days. One ring was selected and inoculated into a 1 L baffled shake flask containing 250 mL YNB and cultivated at 30 °C, 220 rpm for 24 h in the shake cultivation. The initial OD600 of the fermenter was 0.8 after inoculation. Fermentation was carried out in 5 L bioreactors (T&J Bioengineering, Shanghai, China). The bioreactor is equipped with dissolved oxygen (DO), pH and temperature measurement probes, and is fed by an automatically controlled feeding system. Fermentation was carried out at 30 °C for 144 h, and reactors initially contained 2.5 L of mineral medium. One litre of medium contained 5 g KH2PO4, 8 g (NH4)2SO4, 0.8 g MgSO4·7H2O, 40 g d-glucose, 1.5 mL vitamin solution and 2.5 mL trace metal solution (Sun et al., 2021). When initial glucose was depleted, the glucose concentration was maintained at 0.5 g/L by adding 800 g/L glucose in the fermenter. When optimizing pH, it was maintained between 3.5 and 6.5 by adding 8 M KOH while DO was maintained at 10 % by controlling agitation and ventilation and the residual sugar concentration was maintained at 0.5 g/L by adding 800 g/L glucose. When optimizing DO, it was maintained between 0 % and 30 % by controlling agitation and ventilation while pH was maintained at 4.5 and the residual glucose concentration was maintained at 0.5 g/L. While optimizing the carbon/nitrogen (C/N) ratio, C/N was maintained between 5:1 and 20:1 by feeding pumps with a flow of well-proportioning (NH)2SO4 and glucose while DO was maintained at 20 %, pH was maintained at 4.5, and the residual sugar concentration was maintained at 0.5 g/L. Samples were taken three times every 8 h to measure OD600, dry cell weight (DCW), glucose concentration, p-CA and resveratrol. All the samples were conducted in triplicate.    3.1. Construction of a basal resveratrol-producing strain\n",
       "For the efficient synthesis of resveratrol, a de novo resveratrol biosynthesis pathway was introduced in Y. lipolytica PO1f (Fig. 1A). Six resveratrol synthesis pathways were constructed by combining six genes (FjTAL from Flavobacterium johnsoniaeu; RgTAL from Rhodotorula glutinis; TcTAL from Trichosporon cutaneum; Pc4CL1 from Petroselinum crispum; At4CL1 from Arabidopsis thaliana; VvSTS from Vitis vinifera) from different sources with the strong TEF promoter. Accumulation of p-CA and resveratrol was detected in all engineered strains (Fig. 1B). The resveratrol titer ranged from 0.3 mg/L to 1.4 mg/L after 96 h of fermentation in 24 deep-well plates. The resveratrol titer of FjTAL strains ST413 and ST414 was slightly higher than that of strains containing RgTAL and TcTAL. To investigate whether the precursor l-tyrosine limits the synthesis of resveratrol, 200 mg/L of l-Tyr precursor was added for fermentation, and this increased resveratrol accumulation in all engineered strains, among which strain ST413 (FjTAL-Pc4CL1-VvSTS) reached the highest (4.6 mg/L) (Fig. 1C). Therefore, ST413 was used as the initial strain for subsequent strain construction. These results showed that the combination of FjTAL and Pc4CL1 achieved the highest resveratrol titer. Fig. 1. Biosynthetic pathways of resveratrol in engineered Y. lipolytica. (A) G-6-P, glucose-6-phosphate; F-6-P, fructose-6-phosphate; F-1,6BP, fructose-1,6-bisphosphate； PEP, phosphoenolpyruvate; E4P, erythrose-4-phosphate; DAHP, 3-deoxy-arabino-heptulonate-7-phosphate; EPSP, 5-enolpyruvylshikimate 3-phosphate; CHA, chorismic acid; l-Phe, l-phenylalanine; l-Tyr, l-tyrosine; ARO3, 3-deoxy-7-phosphoheptulonate synthase; ARO4, 3-deoxy-7-phosphoheptulonate synthase; ARO1, pentafunctional AROM polypeptide; ARO7, Chorismite mutase; FPK, phosphoketolase; PTA, phosphotransferase; TAL, tyrosineammonia lyase; 4CL, 4-coumarate-CoA ligase; STS, resveratrol synthase; PAL, phenylalanine ammonia lyase; C4H, trans-cinnamate 4-monooxygenase; AtATR2, threonine-protein kinase ATR2; YlCYB5, cytochrome b5; DGA1, diacylglycerol O-acyltransferase 1; TAG, triacylglycerol; fbr, feedback-resistant. (B) The titer of p-CA and resveratrol without l-tyrosine. (C) The titer of p-CA and resveratrol with 200 mg/L l-tyrosine (l-Tyr).\n",
       "\n",
       "3.2. Engineering the shikimic acid (SA) pathway for improved resveratrol biosynthesis\n",
       "Addition of l-Tyr showed that enhancing the supply of this precursor could increase the efficiency of resveratrol biosynthesis. Therefore, removing feedback inhibition may be an effective method to increase l-Tyr. According to previous studies, the mutant alleles ARO4 and ARO7 are common engineering targets for enhancing l-Tyr to resveratrol production in Y. lipolytica (Saez-Saez et al., 2020) and S. cerevisiae (Li et al., 2016). Allelic mutant genes Aro4K221L and Aro7G139S were constructed under the strong TEF promoter, and the resveratrol titer was significantly increased when either allele was expressed (Fig. 2). Furthermore, when both Aro4K221L and Aro7G139S were expressed in strain ST420 (FjTAL-PC4CL1-VvSTS, Aro4K221L-Aro7G139S), the titer reached 96.2 mg/L after 96 h fermentation in 24-deep well plates. An attempt to further enhance l-Tyr accumulation was made by overexpressing intron-removed YlARO1. Based on strain ST420, accumulation of resveratrol and p-CA reached 111.2 mg/L and 131.2 mg/L, respectively, by overexpressing intron-removed YlARO1.\n",
       "\n",
       "\n",
       "3.3. Expression of a Pc4CL1-VvSTS fusion enhances resveratrol biosynthesis\n",
       "Although SA feedback-resistant inhibition showed a significant increase in resveratrol titer, the precursor also resulted in a high accumulation of p-CA. To further improve the titer of resveratrol, fusion proteins of Pc4CL1 and VvSTS were used to reduce the potential loss of unstable intermediates. In order to achieve a strong priming effect, the strong 8xUASTEFin promoter (see Supplementary Materials) joined with eight upstream activating sequence (UAS) motifs was constructed to drive expression of the fusions. Specifically, VvSTS was fused to the C-terminus of Pc4CL1 under the control of the strong 8xUASTEFin promoter with a rigid (TPTP, TPTPx2, EAAAK, EAAAKx2) or flexible (GGGGS, GGGGSx2, GSG, GSGx2) linking peptide. Fusion proteins were expressed in SA feedback-resistant strains. The results showed that partial linkers could increase resveratrol production. The ST427 strain expressing the E3AK fusion was the highest titer of 149.2 mg/L, but accumulation of p-CA was only 6.1 mg/L in this strain (Fig. 3).\n",
       "\n",
       "\n",
       "Fusing Pc4CL1 and VvSTS increased the titer of resveratrol, but only a small amount of p-CA accumulated, suggesting that the supply of precursor was insufficient. To improve precursor supply, the PAL branch pathway was used to enhance the accumulation of p-CA. AtPAL and AtC4H from A. thaliana, and SmPAL and SmC4H from Silybum marianum were introduced and co-expressed with endogenous YlCYB5 and AtATR2 from A. thaliana to construct the PAL branch pathway in strain ST427. The titer of resveratrol was increased to 175.1 mg/L in the ST450 (AtPAL-AtC4H) strain (Fig. 4A). The enzyme DAHP synthase (ARO3) is feedback-inhibited by l-Phe (Cao et al., 2020). By overexpressing the allelic mutant YlARO3K225L, the titer of resveratrol reached 220.1 mg/L in strain ST452, and 50.2 mg/L of p-CA was accumulated (Fig. 4A).\n",
       "\n",
       "Low accumulation of p-CA may be the main factor limiting the resveratrol titer. It was speculated that factors upstream may restrict carbon flux to the SA pathway. In order to increase carbon flow into the SA pathway, phosphoketolase from Bifidobacterium brebifidum (BbFPK), Bifidobacterium adolescentis (BaFPK) and Clostridium acetobutylicum (CaFPK), and phosphotransferase from Bacillus subtilis (BsPTA) and E. coli (EcPTA) were used to construct the phosphoketolase (PHK) pathway. The PHK pathway was introduced into strain ST452 to increase the supply of precursor E4P (Fig. 4B). To much surprise, the titer of resveratrol was not significantly increased, but p-CA accumulation was relatively high. Strain ST484 in which CaFPK and BsPTA were integrated produced 235.1 mg/L resveratrol, as well as 518.3 mg/L p-CA (Fig. 4C), suggesting the resveratrol biosynthesis and malonyl-CoA supply genes should be further optimised.\n",
       "\n",
       "3.5. Multi-copy integration of key genes improves resveratrol biosynthesis\n",
       "Redistributing carbon metabolism leads to the accumulation of p-CA. Herein, a multi-copy integration strategy was used to improve the downstream metabolic pathway. Two multi-copy site sequences have been identified in Y. lipolytica, 26S rDNA (Lv et al., 2019) and ZETA (Bordes et al., 2007), which have a copy number of 100–200 in the genome. Because Y. lipolytica Po1f is URA3- and LEU-deficient, the auxotroph can be recovered following integration, which is conducive to large-scale industrial fermentation. To achieve higher filtering efficiency, weak PLEU promoters were used to construct URA3 and LEU screening tags. Expression of the Pc4CL1-VvSTS fusion was achieved using 26S rDNA carrying URA3 (Fig. 5A). As shown in Fig. 5B, 72 clones were randomly selected for 24 deep-well plate fermentation, and all strains produced resveratrol and p-CA. Following the first round of 26S rDNA random integration, the highest titer was 486.2 mg/L with strain ST888 (Fig. 5B). Since the uracil deficiency was recovered in the previous 26S rDNA integration process, the LEU marker could be recovered by constructing screening tags in the process of ZETA multi-copy integration. Following the second round of ZETA integration in strain ST888, 72 clones were randomly selected for fermentation, and the highest titer of resveratrol was 632.1 mg/L with strain ST889 (Fig. 5C). In order to further increase the titer of resveratrol, it was necessary to increase the supply of malonyl-CoA. By knocking out DGA1, which is responsible for triacylglycerol (TAG) production, in strain ST889, the titer of resveratrol reached 819.1 mg/L with strain ST890 (Fig. 5C).\n",
       "\n",
       "Malonyl-CoA is also positioned at the main rate-limiting step of resveratrol biosynthesis. In a previous study, YlACC1 overexpression was used to increase the supply of malonyl-CoA, but it did not increase the titer of resveratrol (Saez-Saez et al., 2020). In Y. lipolytica, malonyl-CoA is mainly used for the synthesis of TAG, therefore reducing TAG production may be an effective means of accumulating malonyl-CoA (Tai & Stephanopoulos, 2013). Indeed, resveratrol was increased 1.3-fold (819.1 mg/L) compared with ST889 by knocking out DGA1. These results also indicate that simply overexpression related genes of malonyl-CoA may result in the accumulation of TAG, while weakening TAG accumulation is an effective strategy to enhance malonyl-CoA to improve resveratrol production.\n",
       "\n",
       "3.6. Efficient production of resveratrol in a 5 L bioreactor by morphology control\n",
       "Y. lipolytica is a typical dimorphic yeast (Vandermies et al., 2018). In a previous study, filamentous forms are mainly used to resist to a harsh environment (Liu et al., 2021). Therefore, more efficient resveratrol production with strain ST890 was conducted by carefully controlling cell morphology in a 5 L bioreactor. As shown in Fig. 6A, under optimal conditions (pH 4.5), Y. lipolytica existed in both yeast and filamentous forms, and the resveratrol titer reached 11.3 g/L. At pH 4.5, the effect of DO on yeast morphology was explored by controlling the oxygen concentration, and at 20 % DO the filamentous morphology was scarce, most cells were in the yeast form, and the resveratrol titer reached 15.6 g/L (Fig. 6B). At pH 4.5 and 20 % DO, cell morphology was altered by manipulating the feed flow to control the C/N ratio. When the C/N ratio was 10:1, almost all Y. lipolytica cells were in the yeast form, and the resveratrol titer reached the highest level of 22.5 g/L (Fig. 6C) with no more than 20 mg/L of p-CA accumulated. Fig. 6. Fed-batch fermentation optimisation in a 5 L bioreactor. (A) Strain morphology and fermentation results under different pH conditions (DO 10 %). (B) Strain morphology and fermentation results under different DO conditions (pH 4.5). (C) Strain morphology and fermentation results under different C/N conditions (pH 4.5, DO 20 %). (D) Results of fermentation under optimal conditions (pH 4.5, DO 20 %, C/N 10:1).\n",
       "\n",
       "By optimizing pH, DO and C/N in a 5 L fermenter, Y. lipolytica was maintained in the yeast form, a total of 961.8 g glucose was added during the fermentation of 144 h and fed-batch fermentation of ST890 produced 22.5 g/L (0.28 g/g DCW) resveratrol (Fig. 6D), with a yield on glucose of 65.5 mg/g and a productivity of 0.16 g/L/h. Importantly, this high-level synthesis of resveratrol was achieved using only inexpensive mineral fermentation media, which to some extent reduced lipid accumulation and thus increased the accumulation of malonyl-CoA.\n",
       "\n"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#prompt 2\n",
    "display(Markdown(prompt2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "e9a161d4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "A multi-column spreadsheet from the given text  | experiment type | culture volume | culture vessel | substrate | substrate concentration | oxygen condition | nitrogen condition | time | pH | temperature |      2. Materials and methods\n",
       "2.1. Strains and medium\n",
       "The Y. lipolytica Po1f (Lv et al., 2019) was used as the starting host strain. Plasmids were constructed and propagated using E. coli JM109. Cultures of E. coli strains at 37 °C were carried out on lysogeny broth (LB) or agar plates containing ampicillin (100 mg/L). Y. lipolytica was cultured on yeast nitrogen base (YNB) medium (6.74 g/L YNB with or without 0.47 g/L amino acids, 20 g/L glucose) or yeast extract peptone dextrose (YPD) medium (20 g/L glucose, 20 g/L peptone and 10 g/L yeast extract) at 30 °C. For plasmids with LEU tags, Y. lipolytica strains were grown on YNB with uracil dropout. For plasmids with URA tags, Y. lipolytica were grown on YNB with leucine dropout. For the non-deficient defective strain, Y. lipolytica were grown on YNB media.\n",
       "\n",
       "2.2. Construction of vectors and strains\n",
       "The strains and plasmids used in this study are shown in Supplementary Materials. A set of vectors from YaliBricks was used to construct the plasmids needed for genome engineering (Lv et al., 2019). All plasmids were constructed using a Gibson assembly kit (Sangon Biotech, Shanghai, China). Especially, the fusion of Pc4CL1 and VvSTS proteins was constructed by adding homologous arms of linker sequences at both ends. All heterologous genes were codon optimized and synthesized by Sangon Biotech (Sangon Biotech, Shanghai, China). All endogenous genes were amplified from the Y. lipolytica PO1f genome by using a High-fidelity Phusion DNA polymerase (Vazyme, Nanjing, China). All plasmids were confirmed using Sanger sequencing by Sangon Biotech (Sangon Biotech, Shanghai, China). CRISPR-Cas9 system (Wong et al., 2017) was used for single gene integration or knock-out. CRISPR RGEN Tools (https://chopchop.cbu.uib.no/) website was used for the specific guide RNA (sgRNA) design. The transforming plasmids or genome integration of Y. lipolytica PO1f were performed according to the requirements of A Frozen-EZ Yeast Transformation II Kit (Zymol Research, CA, USA). After transformation, the transformed cells were evenly coated on the corresponding selective solid medium and cultivated at 30 °C for 3–5 days.\n",
       "\n",
       "For multi-copy integration, 26S ribosomal DNA (rDNA) and ZETA sequences were copied from the Y. lipolytica genome. The fragments are connected to the upstream and downstream homologous arms of 26S rDNA or ZETA sequence, generating prDNA::(URA3-Pc4CL1-E3AK-VvSTS) and pZETA::(LEU-Pc4CL1-E3AK-VvSTS), respectively. Each vector was linearised to Y. lipolytica transformation.\n",
       "\n",
       "2.3. Strain cultivation and fed-batch fermentation\n",
       "All small-scale fermentation was carried out in 24-well plates. For 24-well plate fermentation, strains were spread on YPD plates and cultivated at 30 °C for 2–3 days. After cultivated, single colony was randomly selected in 24 deep-well plates containing 2 mL of YPD medium and incubated at 30 °C and 220 rpm for 18 h as seed. They were then transferred to new 24-well plates containing 4 mL YPD medium at an initial inoculum of 5 % for fermentation. After 96 h, OD600 was made and samples were collected for high-performance liquid chromatography (HPLC) analysis.\n",
       "\n",
       "For fed-batch fermentation, strains from seed culture preparation were spread on YPD plates and cultivated at 30 °C for 2–3 days. One ring was selected and inoculated into a 1 L baffled shake flask containing 250 mL YNB and cultivated at 30 °C, 220 rpm for 24 h in the shake cultivation. The initial OD600 of the fermenter was 0.8 after inoculation. Fermentation was carried out in 5 L bioreactors (T&J Bioengineering, Shanghai, China). The bioreactor is equipped with dissolved oxygen (DO), pH and temperature measurement probes, and is fed by an automatically controlled feeding system. Fermentation was carried out at 30 °C for 144 h, and reactors initially contained 2.5 L of mineral medium. One litre of medium contained 5 g KH2PO4, 8 g (NH4)2SO4, 0.8 g MgSO4·7H2O, 40 g d-glucose, 1.5 mL vitamin solution and 2.5 mL trace metal solution (Sun et al., 2021). When initial glucose was depleted, the glucose concentration was maintained at 0.5 g/L by adding 800 g/L glucose in the fermenter. When optimizing pH, it was maintained between 3.5 and 6.5 by adding 8 M KOH while DO was maintained at 10 % by controlling agitation and ventilation and the residual sugar concentration was maintained at 0.5 g/L by adding 800 g/L glucose. When optimizing DO, it was maintained between 0 % and 30 % by controlling agitation and ventilation while pH was maintained at 4.5 and the residual glucose concentration was maintained at 0.5 g/L. While optimizing the carbon/nitrogen (C/N) ratio, C/N was maintained between 5:1 and 20:1 by feeding pumps with a flow of well-proportioning (NH)2SO4 and glucose while DO was maintained at 20 %, pH was maintained at 4.5, and the residual sugar concentration was maintained at 0.5 g/L. Samples were taken three times every 8 h to measure OD600, dry cell weight (DCW), glucose concentration, p-CA and resveratrol. All the samples were conducted in triplicate."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#prompt 3\n",
    "display(Markdown(prompt3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "e618a016",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import openai\n",
    "from IPython.display import Markdown\n",
    "# Load your API key from an environment variable or secret management service\n",
    "openai.api_key = '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'\n",
    "def gpt35(prompt):\n",
    "    response = openai.ChatCompletion.create(\n",
    "        model = \"gpt-3.5-turbo\",\n",
    "        temperature = 0,\n",
    "        presence_penalty = -2,\n",
    "        messages=[{\"role\": \"user\", \"content\": prompt}])\n",
    "    return response['choices'][0]['message']['content']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "8f85958e",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "| Strain | Substrate concentration | Nitrogen source | Culture volume | Culture vessel | Culture time | pH | Temperature |\n",
       "|--------|------------------------|----------------|----------------|----------------|--------------|----|-------------|\n",
       "| JBEI-3695 | Not specified | Not specified | Not specified | Not specified | Not specified | Not specified | Not specified |\n",
       "| Not specified | Not specified | Not specified | Not specified | Not specified | Not specified | Not specified | Not specified |"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "response_method = gpt35(\"Find the strain, substrate concentration, nitrogen source, culture volume, culture vessel, and culture time, pH, temperature, and convert to a table with columns |strain| substrate concentration| nitrogen source| culture volume| culture vessel| and culture time| pH| temperature| in the given text:      \"+Method_text)\n",
    "display(Markdown(response_method))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "3476a8cc",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/markdown": [
       "I'm sorry, as an AI language model, I do not have the capability to create spreadsheets. However, I can provide you with the steps to create spreadsheets using Microsoft Excel or Google Sheets. \n",
       "\n",
       "To create a spreadsheet using Microsoft Excel, follow these steps:\n",
       "\n",
       "1. Open Microsoft Excel on your computer.\n",
       "2. Click on the \"Blank workbook\" option to create a new spreadsheet.\n",
       "3. Enter your data into the spreadsheet, using the cells to organize your information.\n",
       "4. Use the \"Format\" option to adjust the appearance of your spreadsheet, including font, color, and alignment.\n",
       "5. Use the \"Formulas\" option to create calculations and formulas to analyze your data.\n",
       "6. Save your spreadsheet by clicking on \"File\" and then \"Save\" or \"Save As\" to create a new file.\n",
       "\n",
       "To create a spreadsheet using Google Sheets, follow these steps:\n",
       "\n",
       "1. Open Google Sheets on your computer.\n",
       "2. Click on the \"Blank\" option to create a new spreadsheet.\n",
       "3. Enter your data into the spreadsheet, using the cells to organize your information.\n",
       "4. Use the \"Format\" option to adjust the appearance of your spreadsheet, including font, color, and alignment.\n",
       "5. Use the \"Formulas\" option to create calculations and formulas to analyze your data.\n",
       "6. Save your spreadsheet by clicking on \"File\" and then \"Save\" or \"Save As\" to create a new file."
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "response_table = gpt35(\"Convert these into spreadsheets:      \"+Table_text)\n",
    "display(Markdown(response_table))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "e47b8395",
   "metadata": {},
   "outputs": [],
   "source": [
    "#"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "89d5cea3",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
